### PR TITLE
Fix link and add class uml diagrams to documentation

### PR DESCRIPTION
**Pull request by kohnh**

### CODE REVIEW

**Comment by owenl131** [comment]
Can add a unit test for the case when module is specified twice in the link and lessontype is given twice in the link?

File: docs/UserGuide.md [hunk 1]
```diff
@@ -806,11 +806,15 @@ Format: `import [ NUSMOD_LINK ]`
   * `LAB`  representing LABORATORY
   * `PROJ` representing MINI_PROJECT
   * `SEM`  representing SEMINAR_STYLE_MODULE_CLASS  
-  note: other lesson types are currently not supported.
+  note: other lesson types are currently not supported. Preliminary checks based on the first 4 characters have been implemented to ensure that 
+  * `LESSON_TYPE_SHORT_FORM` is of the above-mentioned form. E.g. `TUTA` will be accepted as representing TUTORIAL.
 * `LESSON_NUMBER` can vary and is not of a certain form. Take note `01` is not the same as `1`. 
 * If `SEMESTER_NUMBER` is incorrect, the whole link will not be parsed.
 * If `MODULE_CODE` is incorrect, that module will not be added.
+* If there are duplicated `MODULE_CODE`, only the first module information will be parsed.
```
Comment (owenl131) [inline docs/UserGuide.md:814]: Unit test for this

File: docs/UserGuide.md [hunk 2]
```diff
@@ -806,11 +806,15 @@ Format: `import [ NUSMOD_LINK ]`
...
 * If `LESSONS_INFO` is incorrect, that lesson will not be added.
+* If there are duplicated `LESSON_TYPE` within the lesson information, only the last lesson 
+information of that `LESSON_TYPE` will be saved although all will be parsed.
```
Comment (owenl131) [inline docs/UserGuide.md:817]: and this

**Review by owenl131 — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 1 above)
Comment (kohnh) [reply docs/UserGuide.md:816]: Added

File: docs/UserGuide.md (hunk 2 above)
Comment (kohnh) [reply docs/UserGuide.md:819]: Added

**Review by CheahHaoYi — APPROVED** [review]
Looks good to merge!

**Pull request merged**